lib/amqp/client/connection.rb

```diff
@@ -443,7 +443,7 @@ def establish(socket, user, password, vhost, options)
 
           type, channel_id, frame_size = buf.unpack("C S> L>")
           frame_end = buf.getbyte(frame_size + 7)
-          raise UnexpectedFrameEndError, frame_end if frame_end != 206
+          raise UnexpectedFrameEnd, frame_end if frame_end != 206
```

Comment:
Actually, this is still not correct, it should be `Error::UnexpectedFrameEnd`, noticed this when writing a test for this, I will push it to this branch soon